lib/errors/error_record.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class ErrorRecord
+  attr_reader :error, :sentry_extra, :level
+  attr_writer :level
+
+  def initialize(error, sentry_extra, update_course_stats)
+    @error = error
+    @sentry_extra = sentry_extra
+    @update_course_stats = update_course_stats
+    @level = nil
+  end
+
+  def course
```

Comment:
I think we don't need to care about a `course` here at all now. Basically, it should be the job of the update service to figure out what error reporting behavior is wanted... and the pattern ought to apply to any kind of update, not just an `UpdateCourseStats`. We'll just start with the course updates, but this pattern may be useful for other updates as well.

Reply:
I have made `ErrorRecord` more general and now the specific tasks of error handling is a part of the update object itself.